src/app/qgsapplayertreeviewmenuprovider.cpp

```diff
@@ -314,51 +314,79 @@ QMenu *QgsAppLayerTreeViewMenuProvider::createContextMenu()
 
           // SQL dialog
           if ( conn->capabilities().testFlag( QgsAbstractDatabaseProviderConnection::Capability::ExecuteSql ) )
-            menu->addAction( QgsApplication::getThemeIcon( QStringLiteral( "/dbmanager.svg" ) ), tr( "Execute SQL…" ), menu, [layer, this] {
-              std::unique_ptr<QgsAbstractDatabaseProviderConnection> conn2 { QgsMapLayerUtils::databaseConnection( layer ) };
-              if ( conn2 )
+          {
+            QAction *executeSqlAction = new QAction( QgsApplication::getThemeIcon( QStringLiteral( "/dbmanager.svg" ) ), tr( "Execute SQL…" ), menu );
```

Comment:
TODO one day, copy/rename that icon to be slightly more generic than referring to DB Manager as we move forward to phasing that out.